src/walletConnect/v2/saga.ts

```diff
@@ -0,0 +1,124 @@
+import '@react-native-firebase/database'
+import '@react-native-firebase/messaging'
+import SignClient from '@walletconnect/sign-client'
+import { SignClientTypes } from '@walletconnect/types'
+import { getSdkError } from '@walletconnect/utils'
+import { EventChannel, eventChannel } from 'redux-saga'
+import { call, put, take, takeLeading } from 'redux-saga/effects'
+import { WalletConnectPairingOrigin } from 'src/analytics/types'
+import Logger from 'src/utils/Logger'
+import {
+  Actions,
+  clientInitialised,
+  initialiseClient,
+  initialisePairing,
+  sessionCreated,
+  sessionDeleted,
+  sessionPayload,
+  sessionProposal,
+  sessionUpdated,
+  WalletConnectActions,
+} from 'src/walletConnect/v2/actions'
+import networkConfig from 'src/web3/networkConfig'
+
+let client: SignClient | null = null
+
+const TAG = 'WalletConnect/saga'
+
+function* handleInitialiseWalletConnect() {
+  const walletConnectChannel: EventChannel<WalletConnectActions> = yield call(
+    createWalletConnectChannel
+  )
+  while (true) {
+    const message: WalletConnectActions = yield take(walletConnectChannel)
+    yield put(message)
+  }
+}
+
+function* createWalletConnectChannel() {
+  if (!client) {
+    Logger.debug(TAG + '@createWalletConnectChannel', `init start`)
+
+    client = yield call([SignClient, 'init'], {
+      logger: 'debug',
+      // TODO get proper project ID - having issues setting up a valora
+      // walletconnect account
+      // is this a secret or env variable?
+      // https://docs.walletconnect.com/2.0/advanced/relay-server
+      projectId: '906f08083218680fedb1502f372b0b35',
+      relayUrl: networkConfig.walletConnectEndpoint,
+    })
```

Comment:
Should we also add the [`metadata`](https://docs.walletconnect.com/2.0/javascript/guides/react-native#create-session)?

Reply:
good shout! added